sdk/eventhub/Azure.Messaging.EventHubs/src/Consumer/ReadEventOptions.cs

```diff
@@ -84,6 +90,82 @@ public class ReadEventOptions
             }
         }
 
+        /// <summary>
+        ///   The number of events that will be read from the Event Hubs service and held in a local memory
+        ///   cache when reading is active and events are being emitted to an enumerator for processing.
+        /// </summary>
+        ///
+        /// <value>
+        ///   The <see cref="CacheEventCount" /> is a control that developers can use to help tune performance for the specific
+        ///   needs of an application, given its expected size of events, throughput needs, and expected scenarios for using
+        ///   Event Hubs.
+        /// </value>
+        ///
+        /// <remarks>
+        ///   The size of this cache has an influence on the efficiency of reading events from the Event Hubs service.  The
+        ///   larger the size of the cache, the more efficiently service operations can be buffered in the background to
+        ///   improve throughput.  This comes at the cost of additional memory use and potentially increases network I/O.
+        ///
+        ///   For scenarios where the size of events is small and many events are flowing through the system, using a larger
+        ///   <see cref="CacheEventCount"/> and <see cref="PrefetchCount" /> may help improve throughput.  For scenarios where
+        ///   the size of events is larger or when processing of events is expected to be a heavier and slower operation, using
+        ///   a smaller size <see cref="CacheEventCount"/> and <see cref="PrefetchCount"/> may help manage resource use without
+        ///   incurring a non-trivial cost to throughput.
+        ///
+        ///   Regardless of the values, it is generally recommended that the <see cref="PrefetchCount" /> be at least 2-3
+        ///   times as large as the <see cref="CacheEventCount" /> to allow for efficient buffering of service operations.
+        /// </remarks>
+        ///
+        public int CacheEventCount
```

Comment:
Outside of sharing code, not really.  There's no "is-a" relationship, so I'd prefer to avoid an artificial hierarchy for code sharing purposes.   (it would be nice to have a centralized dictionary of doc comments to reference, but I digress....)

Reply:
There's always the `<include>` tag 😄 
https://docs.microsoft.com/en-us/dotnet/csharp/codedoc#include

Reply:
Now that's just evil!  😛